packages/permissions/src/hooks/use-is-authorized/use-is-authorized.spec.tsx

```diff
@@ -323,230 +407,277 @@ describe('when only one permission matches', () => {
             },
           });
 
-          expect(queryByText('Is authorized: Yes')).toBeInTheDocument();
+          expect(
+            rendered.queryByText('Is authorized: Yes')
+          ).toBeInTheDocument();
         });
       });
-      describe('when dataFence permission is not met', () => {
-        it('should indicate as authorized', () => {
-          const { queryByText } = testRender({
-            allAppliedPermissions: [{ name: 'canManageOrders', value: true }],
-            allAppliedDataFences: [
-              {
-                __typename: 'StoreDataFence',
-                type: 'store',
-                group: 'orders',
-                name: 'canViewOrders',
-                value: 'store-1',
-              },
-            ],
-            demandedPermissions: ['ManageOrders'],
-            demandedDataFences: [
-              {
-                type: 'store',
-                group: 'orders',
-                name: 'ManageOrders',
-              },
-            ],
-            selectDataFenceData: ({ type }) => {
-              switch (type) {
-                case 'store':
-                  return ['store-1'];
-                default:
-                  return null;
-              }
+    });
+    describe('when overwritten', () => {
```

Comment:
This is the new case.

Reply:
I really don't understand this new test, so if:
1- you have datafence permissions
2- you also have a general permission, but not the action right
then `isAuthorized` would return `false`?

Reply:
Let me try:

So the story is you have a data fence configured. However, this data fence is practially not valid. This is cause you have manage access on the resource permission level. The mental modal is: it's additive from general permission to data fences. Not substractive. Saying you can't have: manage orders (resource access/general permission) and take away manage right to only view orders in a specific store. What you have to do instead is to set the resource access to view and then add manage to the stores the team can manage in the data fence.

This means that
Having set manage orders as general permission
Having set manage orders as data fence in e.g. store-1
-> These two mean the same and doesn't impact anything
Having set add discount to false as action right 
-> This now makes it contradicting here the action right and general permission should take precedence. That is cause the data fence itself is not valid in combination with the resource access. You already said: all orders can be managed - no matter the store. There is no need to say store-1 can also manage orders. It's implied by the general permission already.

Does that help? @adnasa does that make sense?

Reply:
We can also have a call if that helps! Just ping me in Slack :)